docs/markdown/01-intro/05-JSX.md

```diff
@@ -0,0 +1,291 @@
+
+<!-- .element: class="important"-->
+
+- This is a **valid** JSX
+
+```jsx
+const element = (
+  <>
+    <img data-cy="img" src="link here" alt="desc here" className="card__img" />
+  </>
+);
+```
+
+- This is **not** a valid JSX
+
+```jsx
+const element = (
+  <>
+    <img src="link here" alt="desc here" class="card__img" />
+  </>
+);
+```
+
+_For historical reasons, `aria-*` and `data-*` attributes are written as in HTML with dashes._
+
+##==##
+
+<!-- .slide: class="transition-bg-blue-3 right"-->
+
+# Javascript in JSX with Curly Braces
+
+##==##
+
+# JSX with Curly Braces
+
+In JSX, you can utilize curly braces to embed JavaScript expressions. This allows for dynamic content within your JSX elements.
+
+<br/>
+
+- **Variables**
+
+```jsx
+const name = 'John Doe';
+
+const element = <p>Hello, {name}!</p>;
+```
+
+- **Functions**
+
+```jsx
+function greet() {
+  return 'Hello, world!';
+}
+
+const element = <p>{greet()}</p>;
+```
+
+- **Terinary operator**
+
+```jsx
+const name = 'John Doe';
+
+const element = <p>{name ? `Hello, ${name}!` : 'Hello, stranger!'}</p>;
+```
+
+##==##
+
+# JavaScript object inside JSX with curly braces
+
+- **Objects**
+
+```jsx
+const user = {
+  firstName: 'John',
+  lastName: 'Doe',
+};
+
+const element = (
+  <p>
+    Hello, {user.firstName} {user.lastName}!
+  </p>
+);
+```
+
+- **CSS style property**
+
+```jsx
+const style = {
+  color: 'red',
+  fontSize: '20px',
+};
+
+const element = <p style={style}>Hello, world!</p>;
+```
+
+- CSS style property **double curlies**
+
+```jsx
+const element = <p style={{ color: 'red', fontSize: '20px' }}>Hello, world!</p>;
+```
+
+<br/>
+
+_The next time you see `{{` and `}}` in JSX, know that it’s nothing more than an object inside the JSX curlies!_
+
+<!-- .element: class="important"-->
+
+##==##
+
+<!-- .slide: class="two-column"-->
+
+# Arrays in JSX
+
+_You can render an array of **JSX elements** in JSX, but you need to add a **`key`** attribute to each element._
+
+- This is **valid** JSX
+<!-- .element: class="important"-->
+
+```jsx
+const names = ['John', 'Jane', 'Jack'];
+
+const element = <ul>{names}</ul>;
+```
+
+- This is **not** a valid JSX
+<!-- .element: class="important"-->
+
+```jsx
+const names = [{ name: 'John' }, { name: 'Jane' }, { name: 'Jack' }];
+
+const element = <ul>{names}</ul>;
+```
+
+- Array of JSX elements with **`key`** attribute. This is **Valid** JSX (but not recommended)
+<!-- .element: class="important"-->
+
+```jsx
+const names = [<li key={1}> John </li>, <li key={2}> Jane </li>, <li key={3}> Jack </li>];
+
+const element = <ul>{names}</ul>;
+```
+
+- This is **valid** JSX (but not recommended)
+<!-- .element: class="important"-->
+
+```jsx
+const names = [{ name: 'John' }, { name: 'Jane' }, { name: 'Jack' }];
+
+const array = [<li key={1}>{names[0].name}</li>, <li key={2}>{names[1].name}</li>, <li key={3}>{names[2].name}</li>];
+
+const element = <ul>{array}</ul>;
+```
+
+##==##
```

Comment:
suggestion: all this part is about the chapter loop in fact, so keep it in a place and let move it in the corresponfign chapter

Reply:
So we don't keep the Array part at all in this chapter? Or should I just move the examples for the section "Arrays with **.map** to turn them into JSX elements" to another chapter?

Reply:
IMO, I think we should leave this part as is, as it only gives an introduction while still being concise enough to not show too much about loops in React. Knowing that we are going to dive deeper into the subject (e.g. components, memo, and composition), we might use this "introduction" to kick-start the subject in the corresponding chapter.

Reply:
ok let's do that